aws/default_retryer.go

```diff
@@ -34,25 +35,39 @@ var seededRand = rand.New(&lockedSource{src: rand.NewSource(time.Now().UnixNano(
 // RetryRules returns the delay duration before retrying this request again
 func (d DefaultRetryer) RetryRules(r *Request) time.Duration {
 	// Set the upper limit of delay in retrying at ~five minutes
-	minTime := 30
+	var minTime int64
+	if d.MinTime != 0 {
+		minTime = d.MinTime
+	} else {
+		minTime = 30
+	}
+
+	var initialDelay time.Duration
 	throttle := d.shouldThrottle(r)
 	if throttle {
-		if delay, ok := getRetryDelay(r); ok {
-			return delay
+		if delay, ok := getRetryAfterDelay(r); ok {
+			initialDelay = delay
 		}
 
 		minTime = 500
```

Comment:
This should use a const type now that the value will be configurable, e.g. `DefaultRetryerThrottleDelay`